app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,64 @@
+class RecipesController < ApplicationController
+  before_action :set_recipe, only: %i[show destroy]
+
+  def index
+    @recipes = Recipe.all
+  end
+
+  def show
+    @recipe = Recipe.find(params[:id])
+    @foods = Food.all
+  end
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def create
+    @recipe = Recipe.new(recipe_params)
+    @recipe.user = current_user
+
+    if current_user.nil?
+      redirect_to new_user_session_path, notice: 'You need to sign in or sign up before continuing.'
+      return
+    end
+
+    if @recipe.save
+      redirect_to recipes_path, notice: 'Recipe was successfully created.'
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @recipe = Recipe.find(params[:id])
+    if current_user == @recipe.user
+      @recipe.destroy
+      redirect_to recipes_path, notice: 'Recipe was successfully destroyed.'
+    else
+      redirect_to recipes_path, notice: 'You are not authorized to delete this recipe.'
+    end
+  end
+
```

Comment:
- [ ] Please make sure that food items can be deleted from the list of foods. As you can see from the video below I can't delete food item :smile: 

[Screencast from 04-07-2023 03:10:39 PM.webm](https://user-images.githubusercontent.com/92303891/230632585-8e87e79e-ad2e-4294-8100-1e09fa117e6c.webm)

Reply:
@Owusu-Desmond Hi, is this part of the requirements?